root/static/scripts/edit/URLCleanup.js

```diff
@@ -4544,6 +4543,35 @@ const CLEANUPS: CleanupEntries = {
     ],
     restrict: [LINK_TYPES.otherdatabases],
   },
+  'ototoy': {
+    match: new RegExp('^(https?://)?([^/]+\\.)?ototoy\\.jp', 'i'),
+    restrict: [LINK_TYPES.downloadpurchase],
+    validate: function (url, id) {
+      const m = /^https:\/\/ototoy\.jp\/(?:(labels)\/\d+|(_\/default\/p)\/\d+|(_\/default\/a)\/\d+)$/.exec(url);
+      if (m) {
+        const suffix = m[1] || m[2] || m[3];
```

Comment:
It is a very good start Two points:

1. Major: It assumes that the URL is clean but there is no `clean` function. It may block entering some URLs that would just need a small cleanup. For example, see the rule `anidb`.

2. Minor: It can be simplified into:

```suggestion
      const m = /^https:\/\/ototoy\.jp\/(labels|_\/default\/[ap])\/\d+$/.exec(url);
      if (m) {
        const suffix = m[1];
```

Reply:
For the major point, I added a `clean` function in 4e2505f40e. I did some quick spot-checks in a JS console with it and it seemed like the function returned proper URLS, so hopefully we're good to go there.

For the minor point, I committed your suggestion as c094be0. Thanks!